### PR TITLE
changing permissions hierarchy as per ossf docs

### PR DESCRIPTION
**Pull request by jaydee029**
For issue #2979

I re-read the [ossf scorecard documentation](https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions) and realized that in an ideal workflow the permissions at the top level shouldn't be anything more than ` contents:read ` or `read-all`. All the other permissions need to be specified at the job level. (In order to pass the ossf checks)

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

**Comment by jaydee029** [comment]
@hydai would like review this ?

File: .github/workflows/labeler.yml [hunk 1]
```diff
@@ -7,6 +7,8 @@ on:
       - reopened
       - closed
 
+permissions:
+  contents: read
```
Comment (hydai) [inline .github/workflows/labeler.yml:11]: This is handled in jobs. Why do we need it again here?

File: .github/workflows/misc-linters.yml [hunk 2]
```diff
@@ -6,6 +6,9 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
   cancel-in-progress: true
 
+permissions:
```
Comment (hydai) [inline .github/workflows/misc-linters.yml:9]: Ditto.

File: .github/workflows/reusable-build-on-alpine-static.yml [hunk 3]
```diff
@@ -8,7 +8,8 @@ on:
         required: true
       release:
         type: boolean
-
+permissions:
```
Comment (hydai) [inline .github/workflows/reusable-build-on-alpine-static.yml:11]: All jobs in this workflow need the `contents: write` permissions. Why do we need this?

File: .github/workflows/reusable-build-on-android.yml [hunk 4]
```diff
@@ -10,7 +10,8 @@ on:
         type: boolean
       upload_asset_url:
         type: string
-
+permissions:
```
Comment (hydai) [inline .github/workflows/reusable-build-on-android.yml:13]: Ditto.

File: .github/workflows/reusable-build-on-fedora.yml [hunk 5]
```diff
@@ -10,7 +10,8 @@ on:
         type: boolean
       upload_asset_url:
         type: string
-
+permissions:
```
Comment (hydai) [inline .github/workflows/reusable-build-on-fedora.yml:13]: Ditto.

File: .github/workflows/reusable-build-on-macos.yml [hunk 6]
```diff
@@ -11,7 +11,8 @@ on:
         required: true
       release:
         type: boolean
-
+permissions:
```
Comment (hydai) [inline .github/workflows/reusable-build-on-macos.yml:14]: Ditto.

File: .github/workflows/reusable-build-on-manylinux.yml [hunk 7]
```diff
@@ -11,7 +11,8 @@ on:
         required: true
       release:
         type: boolean
-
```
Comment (hydai) [inline .github/workflows/reusable-build-on-manylinux.yml:14]: Ditto.

File: .github/workflows/reusable-build-on-ubuntu.yml [hunk 8]
```diff
@@ -11,7 +11,8 @@ on:
         required: true
       release:
         type: boolean
-
```
Comment (hydai) [inline .github/workflows/reusable-build-on-ubuntu.yml:14]: Ditto.

File: .github/workflows/reusable-build-on-windows-msvc.yml [hunk 9]
```diff
@@ -8,7 +8,8 @@ on:
         required: true
       release:
         type: boolean
-
+permissions:
```
Comment (hydai) [inline .github/workflows/reusable-build-on-windows-msvc.yml:11]: Ditto.

File: .github/workflows/reusable-create-source-tarball.yml [hunk 10]
```diff
@@ -9,6 +9,9 @@ on:
       release:
         type: boolean
 
+permissions:
```
Comment (hydai) [inline .github/workflows/reusable-create-source-tarball.yml:12]: Ditto.

File: .github/workflows/static-code-analysis.yml [hunk 11]
```diff
@@ -39,6 +39,8 @@ on:
       - "CMakeLists.txt"
       - "cmake/**"
 
+permissions:
```
Comment (hydai) [inline .github/workflows/static-code-analysis.yml:42]: Ditto.

File: .github/workflows/wasi-testsuite.yml [hunk 12]
```diff
@@ -22,6 +22,9 @@ on:
       - "include/host/wasi/**"
       - "thirdparty/wasi/**"
 
+permissions:
```
Comment (hydai) [inline .github/workflows/wasi-testsuite.yml:25]: Ditto.

**Review by hydai — CHANGES_REQUESTED** [review]

File: .github/workflows/labeler.yml (hunk 1 above)
Comment (jaydee029) [reply .github/workflows/labeler.yml:11]: Right , I initially didn't pay attention, but as I mentioned in the pr message , below is what ossf scorecard say about reward points after security checks.

image

other than that I'll see if some of them redundant and remove them.

Comment (jaydee029) [reply .github/workflows/labeler.yml:11]: Also , projects with highest ossf score have followed this

File: .github/workflows/reusable-build-on-alpine-static.yml [hunk 13]
```diff
@@ -8,7 +8,8 @@ on:
         required: true
       release:
         type: boolean
-
```
Comment (hydai) [inline .github/workflows/reusable-build-on-alpine-static.yml:11]: You should keep an empty line here just like other files.

File: .github/workflows/reusable-build-on-android.yml [hunk 14]
```diff
@@ -10,7 +10,8 @@ on:
         type: boolean
       upload_asset_url:
         type: string
-
```
Comment (hydai) [inline .github/workflows/reusable-build-on-android.yml:13]: Ditto.

File: .github/workflows/reusable-build-on-fedora.yml [hunk 15]
```diff
@@ -10,7 +10,8 @@ on:
         type: boolean
       upload_asset_url:
         type: string
-
```
Comment (hydai) [inline .github/workflows/reusable-build-on-fedora.yml:13]: Ditto.

File: .github/workflows/reusable-build-on-macos.yml [hunk 16]
```diff
@@ -11,7 +11,8 @@ on:
         required: true
       release:
         type: boolean
-
```
Comment (hydai) [inline .github/workflows/reusable-build-on-macos.yml:14]: Ditto.

File: .github/workflows/reusable-build-on-manylinux.yml (hunk 7 above)
Comment (hydai) [inline .github/workflows/reusable-build-on-manylinux.yml:14]: Ditto.

File: .github/workflows/reusable-build-on-ubuntu.yml (hunk 8 above)
Comment (hydai) [inline .github/workflows/reusable-build-on-ubuntu.yml:14]: Ditto.

File: .github/workflows/reusable-build-on-windows-msvc.yml [hunk 17]
```diff
@@ -8,7 +8,8 @@ on:
         required: true
       release:
         type: boolean
-
```
Comment (hydai) [inline .github/workflows/reusable-build-on-windows-msvc.yml:11]: Ditto.

File: .github/workflows/reusable-create-source-tarball.yml [hunk 18]
```diff
@@ -9,6 +9,9 @@ on:
...
+  contents: read
+
```
Comment (hydai) [inline .github/workflows/reusable-create-source-tarball.yml:14]: Why do some of them have an empty line, but the others don't.

**Review by hydai — CHANGES_REQUESTED** [review]

File: .github/workflows/reusable-create-source-tarball.yml (hunk 18 above)
Comment (jaydee029) [reply .github/workflows/reusable-create-source-tarball.yml:14]: I have fixed this. 
Thank you

**Comment by jaydee029** [comment]
@hydai are windows and windows-msvc workflows intact, or an error is expected ?

**Comment by hydai** [comment]
> @hydai are windows and windows-msvc workflows intact, or an error is expected ?

We knew that, and it's not related to your PR.

**Review by hydai — APPROVED** [review]

**Comment by jaydee029** [comment]
> > @hydai are windows and windows-msvc workflows intact, or an error is expected ?
> 
> We knew that, and it's not related to your PR.

I mean I was asking , if it gives an error

also I think a few workflow files are still without permissions, i another PR would be required.

**Pull request merged**